Controller/Advice/Call.php

```diff
@@ -0,0 +1,255 @@
+
+        $this->api->initSdk();
+        $this->logger->log('Riskified Advise Call building json data from quote id: ' . $quoteId);
+        $this->adviceBuilder->build($params);
+        $callResponse = $this->adviceBuilder->request();
+        $status = $callResponse->checkout->status;
+        $authType = $callResponse->checkout->authentication_type->auth_type;
+        $this->logger->log('Riskified Advise Call Response status: ' . $status);
+        //use this status while backend order validation
+        $this->session->setAdviceCallStatus($status);
+        if($status != "captured"){
+            $adviceCallStatus = 3;
+            $paymentDetails = array(
+                'date' => $currentDate = date('Y-m-d H:i:s', time()),
+                'auth_type' => 'fraud',
+                'status' => $status
+            );
+            $logMessage = 'Quote ' . $quoteId . ' is set as denied and sent to Riskified. Additional data saved in database (paymentQuote table). Riskified verification (advise call) level.';
+            //saves advise call returned data in quote Payment (additional data)
+            $this->updateQuotePaymentDetailsInDb($quoteId, $paymentDetails);
+            //Riskified defined order as fraud - order data is send to Riskified
+            $this->sendDeniedOrderToRiskified($quote);
+            $this->logger->log($logMessage);
+            $message = 'Checkout Declined.';
+            $this->messageManager->addError(__("Checkout Declined"));
+        }else {
+            $paymentDetails = array(
+                'date' => $currentDate = date('Y-m-d H:i:s', time()),
+                'auth_type' => $authType,
+                'status' => $status
+            );
+            if($authType == "sca"){
+                $adviceCallStatus = false;
+                $message = 'Transaction type: sca.';
+            }else{
+                $adviceCallStatus = true;
+                $message = 'Transaction type: tra.';
+            }
+            //saves advise call returned data in quote Payment (additional data)
+            $this->updateQuotePaymentDetailsInDb($quoteId, $paymentDetails);
+        }
+
+        return  $this->resultJsonFactory->create()->setData(['advice_status' => $adviceCallStatus, 'message' => $message]);
+    }
+
+    /**
+     * Saves quote payment details (additional data).
+     * @param $quoteId
+     * @param $paymentDetails
+     * @throws \Exception
+     */
+    protected function updateQuotePaymentDetailsInDb($quoteId, $paymentDetails)
+    {
+        $quote = $this->registry->registry($quoteId);
+        if (isset($quote)) {
+            $this->logger->log('Quote ' . $quoteId . ' found - saving Riskified Advise or 3D Secure Response as additional quotePayment data in db.');
+            $quotePayment = $quote->getPayment();
+            $additionalData = $quotePayment->getAdditionalData();
+            //avoid overwriting quotePayment additional data
+            if (is_array($additionalData)) {
```

Comment:
if the additionalData is not an array, just declare it as array... no need for this whole duplicated code.